camel-k-runtime-yaml/src/main/java/org/apache/camel/k/yaml/converter/ChoiceConverter.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.k.yaml.converter;
+
+import java.util.List;
+
+import org.apache.camel.model.ChoiceDefinition;
+import org.apache.camel.model.ProcessorDefinition;
+import org.apache.camel.model.WhenDefinition;
+
+public class ChoiceConverter implements StepConverter {
+    @Override
+    public ProcessorDefinition<?> toProcessor(Context context) {
+
+        final Definition definition = context.node(Definition.class);
+        final ChoiceDefinition choice = new ChoiceDefinition();
+
+        ConverterSupport.notNull(definition.when,"when");
+
+        for (Definition.When whenDefinition : definition.when) {
+            ConverterSupport.notNull(whenDefinition.getExpression(), "when.expression");
+            ConverterSupport.notNull(whenDefinition.steps, "when.steps");
+
+            ConverterSupport.convertSteps(
+                context,
+                choice.when().expression(whenDefinition.getExpression()),
+                whenDefinition.steps
+            );
+        }
+
+        if (definition.otherwise != null) {
+            ConverterSupport.notNull(definition.otherwise.steps, "otherwise.steps");
+
+            ConverterSupport.convertSteps(
+                context,
+                choice.otherwise(),
+                definition.otherwise.steps)
+            ;
```

Comment:
Probably CS issues, can you run mvn -P sourcecheck

Reply:
don't think we have check style at the moment :)